typescript/src/bitcoin.ts

```diff
@@ -389,13 +389,21 @@ export function encodeToBitcoinAddress(
 }
 
 /**
- * Decodes P2PKH or P2WPKH address into a public key hash.
+ * Decodes P2PKH or P2WPKH address into a public key hash. Throws if the
+ * provided address is not PKH-based.
  * @param address - P2PKH or P2WPKH address that will be decoded.
  * @returns Public key hash decoded from the address. This will be an unprefixed
  *        hex string (without 0x prefix).
  */
 export function decodeBitcoinAddress(address: string): string {
   const addressObject = new bcoin.Address(address)
+
+  const isPKH =
+    addressObject.isPubkeyhash() || addressObject.isWitnessPubkeyhash()
+  if (!isPKH) {
+    throw new Error("address must be P2PKH or P2WPKH")
```

Comment:
Non-blocking:

We usually start the error text with a capital letter in this project. 

Remember to also update the tests if you decide to fix that.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/510/commits/b446878f9dbe0a8bfaaf89b7d2c6508cb221d9f1